specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2020-05-01-preview/ResourceHealth.json

```diff
@@ -0,0 +1,856 @@
+        }
+      },
+      "required": [
+        "value"
+      ],
+      "description": "The List impactedResourceStatus operation response."
+    },
+    "impactedResourceStatus": {
+      "type": "object",
+      "description": "impactedResource with health status",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "type": "object",
+          "description": "Properties of impacted resource status.",
+          "properties": {
+            "availabilityState": {
+              "type": "string",
+              "description": "Impacted resource status of the resource.",
+              "enum": [
+                "Available",
+                "Unavailable",
+                "Degraded",
+                "Unknown"
+              ],
+              "x-ms-enum": {
+                "name": "AvailabilityStateValues",
+                "modelAsString": true
+              }
+            },
+            "title": {
+              "type": "string",
+              "description": "Title description of the impacted resource status."
+            },
+            "summary": {
+              "type": "string",
+              "description": "Summary description of the impacted resource status."
+            },
+            "reasonType": {
+              "type": "string",
+              "description": "When the resource's availabilityState is Unavailable, it describes where the health impacting event was originated.",
+              "enum": [
+                "Unplanned",
+                "Planned",
+                "UserInitiated"
+              ],
+              "x-ms-enum": {
+                "name": "ReasonTypeValues",
+                "modelAsString": true
+              }
+            },
+            "occuredTime": {
+              "type": "string",
+              "description": "Timestamp for when last change in health status occurred.",
+              "format": "date-time"
+            }
+          }
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "availabilityStatusListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/availabilityStatus"
+          },
+          "description": "The list of availabilityStatuses."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URI to fetch the next page of availabilityStatuses. Call ListNext() with this URI to fetch the next page of availabilityStatuses."
+        }
+      },
+      "required": [
+        "value"
+      ],
+      "description": "The List availabilityStatus operation response."
+    },
+    "availabilityStatus": {
+      "type": "object",
+      "description": "availabilityStatus of a resource.",
+      "properties": {
+        "id": {
```

Comment:
>"id" [](http://example.com/codeflow?start=8&length=4)

What is the motivation for using a reference to resource definition from common (ln 510) vs copying id, name, type and location? consider using trackedResource from definition.

Reply:
This PR is to resolve the S360 Swagger KPI, due to the timeboud and delay, we just tried to follow the same format as our other current API versions and make it consistent everywhere